js/src/common/Model.ts

```diff
@@ -111,6 +112,19 @@ export default abstract class Model {
 
     if ('attributes' in data) {
       this.data.attributes ||= {};
+
+      // @deprecated
+      // Filter out relationships that got in by accident.
+      for (const key in data.attributes) {
```

Comment:
Could this be better as `Object.entries(data.attributes).map(([key, val]) => {})`?

Reply:
I usually prefer functional transformations, but in this case since we're deleting attributes I think an imperative iteration is better.